internal/testing/setup/setup.go

```diff
@@ -100,15 +101,19 @@ func awsSession(region string, client *http.Client) (*session.Session, error) {
 // from files. rf is a modifier function that will be invoked with the address
 // of the httpreplay.Recorder object used to obtain the client; this function
 // can mutate the recorder to add provider-specific header filters, for example.
-func NewRecordReplayClient(ctx context.Context, t *testing.T, rf func(r *httpreplay.Recorder), opts ...option.ClientOption) (c *http.Client, cleanup func()) {
+// An initial state is returned for tests that need a state to have
```

Comment:
I'd add more about the semantics of this state in this comment, it's not very clear. Also add the same for NewAWSSession2

Reply:
Done.